tests/testutil.hpp

```diff
@@ -67,6 +67,9 @@
 #define PORT_6 5561
 
 //  For tests that mock ZMTP
+const uint8_t zmtp_greeting_unversioned[] = {
+  0x01, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
when connecting to a sub, something like this should be a pub message that contains "A":

01 00 02 00 42

Reply:
https://rfc.zeromq.org/spec/13/

Reply:
Yes, I read the RFC, and it is somewhat unclear, at least to me:

> A peer SHALL send a greeting consisting of either an anonymous or an identity message. An anonymous greeting consists of an empty string.

I've tried different combinations for zmtp_greeting_unversioned (including the one you suggested), but none of them connect, and not sure where to go from here.

Reply:
interesting - did you have an old zmq 2.x binary you were doing the connection tests with? Would be easier to look at wireshark I think

Reply:
That's my point -- I don't have an old binary to test with, and I'm not eager to try to build one ;-)  I thought you or @somdoron might be able to tell me.

In the meantime, I've tried looking at the code and RFC to try something that works, but no joy so far:

```
const uint8_t zmtp_greeting_unversioned[] = {
  0x01, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
const uint8_t zmtp_greeting_unversioned[] = {
  0x01, 00, 02, 00, 42};
const uint8_t zmtp_greeting_unversioned[] = {
  0x01, 00};
```

Can anyone tell me what an unversioned handshake looks like?

Reply:
I think nobody is here who knows it off the top of their head - there's a test for the curve protocol that looks like it's attempting to do one: https://github.com/zeromq/libzmq/blob/master/tests/test_security_curve.cpp#L215

Reply:
I will try to make a test for the unversioned protocol

Reply:
Some history: 
https://github.com/zeromq/libzmq/commit/1bca4f6f033ed6bb56fcfd7762bbfea04ecf684b

Reply:
Source code of stream_engine before versioned protocol:
https://github.com/zeromq/libzmq/blob/a622328a9c089e635e8af6b67f6a87251d3e34bf/src/stream_engine.cpp

The bottom line, there is no greeting with the unversioned ZMTP, you just send messages and the first message must be an identity. 

So {0,0} (empty message) is a valid message. Also {1,0,1} (one byte with value 1). 

Because first message must be an identity, to fake an unversioned peer send the following {0,0,1,0,1}, which sends an empty identity following by a single-byte message.

Reply:
The bottom line, I would use dealer to dealer for testing, and then the expect one message to arrive on the zeromq dealer.
The message should be {0,0,1,0,1} (empty identity and single byte message)

Reply:
> I will try to make a test for the unversioned protocol

Hi @somdoron -- I've been off on vacation, so just checking in to see if you are still planning to take a crack at this?